tests/Transforms/new_layout_propagation/insert_conversion.mlir

```diff
@@ -0,0 +1,68 @@
+// RUN: heir-opt --new-layout-propagation --fold-convert-layout-into-assign-layout %s | FileCheck %s
+
+!tensor = tensor<32x32xi16>
+!tensor2 = tensor<32xi16>
+!stensor = !secret.secret<!tensor>
+!stensor2 = !secret.secret<!tensor2>
+
+// Test that when an operation changes the tensor layour in an incompatible way,
+// a layout conversion operation is inserted.
+
+// This is the reduced layout rm_layout1
+#new_layout = #tensor_ext.new_layout<domainSize=1, localSize=4, relation="(d0, d1, d2, d3, d4, d5, d6) : (d0 - d3 * 1024 - d4 == 0, -d1 + d3 == 0, d2 - d5 * 1024 - d6 == 0, d4 - d6 == 0, -d4 + 1023 >= 0, d4 >= 0, -d2 + d5 * 1024 + 1023 >= 0, d2 - d5 * 1024 >= 0)">
+
+// This is the row major layout? rm_layout
+#new_layout1 = #tensor_ext.new_layout<domainSize=2, localSize=4, relation="(d0, d1, d2, d3, d4, d5, d6, d7) : (d0 * 32 + d1 - d4 * 1024 - d5 == 0, -d2 + d4 == 0, d3 - d6 * 1024 - d7 == 0, d5 - d7 == 0, -d5 + 1023 >= 0, d5 >= 0, -d3 + d6 * 1024 + 1023 >= 0, d3 - d6 * 1024 >= 0)">
+
+// This is the column major layout? cm_layout
+#new_layout2 = #tensor_ext.new_layout<domainSize=1, localSize=4, relation="(d0, d1, d2, d3, d4, d5, d6) : (d0 * 32 - d3 * 1024 - d4 == 0, -d1 + d3 == 0, d2 - d5 * 1024 - d6 == 0, d4 - d6 == 0, -d4 + 1023 >= 0, d4 >= 0, -d2 + d5 * 1024 + 1023 >= 0, d2 - d5 * 1024 >= 0)">
+
+// CHECK-DAG: [[rm_layout:[^ ]*]] = #tensor_ext.new_layout<domainSize=2, localSize=4, relation="(d0, d1, d2, d3, d4, d5, d6, d7) : (d0 * 32 + d1 - d4 * 1024 - d5 == 0, -d2 + d4 == 0, d3 - d6 * 1024 - d7 == 0, d5 - d7 == 0, -d5 + 1023 >= 0, d5 >= 0, -d3 + d6 * 1024 + 1023 >= 0, d3 - d6 * 1024 >= 0)">
```

Comment:
As mentioned, we can avoid checking the content of the layout here and defer to unit testing. OK to do that in this PR or next.

Reply:
Currently adding unit testing!